src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests_ListPatterns.cs

```diff
@@ -41,642 +44,76 @@ void verify()
-                }
-                N(SyntaxKind.IsKeyword);
-                N(SyntaxKind.RecursivePattern);
-                {
-                    N(SyntaxKind.LengthPatternClause);
-                    {
-                        N(SyntaxKind.OpenBracketToken);
-                        N(SyntaxKind.RecursivePattern);
-                        {
-                            N(SyntaxKind.LengthPatternClause);
-                            {
-                                N(SyntaxKind.OpenBracketToken);
-                                N(SyntaxKind.DiscardPattern);
-                                {
-                                    N(SyntaxKind.UnderscoreToken);
-                                }
-                                N(SyntaxKind.CloseBracketToken);
-                            }
-                        }
-                        N(SyntaxKind.CloseBracketToken);
-                    }
-                }
-            }
-            EOF();
-        }
-
-        [Fact]
-        public void LengthPattern_15()
-        {
-            UsingExpression(@"c is []",
-                // (1,7): error CS8504: Pattern missing
-                // c is []
-                Diagnostic(ErrorCode.ERR_MissingPattern, "]").WithLocation(1, 7));
-
-            N(SyntaxKind.IsPatternExpression);
-            {
-                N(SyntaxKind.IdentifierName);
-                {
-                    N(SyntaxKind.IdentifierToken, "c");
-                }
-                N(SyntaxKind.IsKeyword);
-                N(SyntaxKind.RecursivePattern);
-                {
-                    N(SyntaxKind.LengthPatternClause);
+                    N(SyntaxKind.ListPattern);
                     {
                         N(SyntaxKind.OpenBracketToken);
-                        M(SyntaxKind.ConstantPattern);
+                        N(SyntaxKind.ListPattern);
                         {
-                            M(SyntaxKind.IdentifierName);
-                            {
-                                M(SyntaxKind.IdentifierToken);
-                            }
+                            N(SyntaxKind.OpenBracketToken);
+                            N(SyntaxKind.CloseBracketToken);
                         }
                         N(SyntaxKind.CloseBracketToken);
                     }
                 }
+                EOF();
             }
-            EOF();
         }
 
         [Fact]
-        public void LengthPattern_16()
+        public void ListPattern_01()
         {
-            UsingExpression(@"c is [0]()",
-                    // (1,1): error CS1073: Unexpected token '('
-                    // c is [0]()
-                    Diagnostic(ErrorCode.ERR_UnexpectedToken, "c is [0]").WithArguments("(").WithLocation(1, 1));
+            UsingExpression(@"c is [[],] v");
+            verify();
 
-            N(SyntaxKind.IsPatternExpression);
+            UsingExpression(@"c is [[],] v", TestOptions.Regular9,
```

Comment:
>Regular9 [](http://example.com/codeflow?start=57&length=8)

I think most of the `Regular9`s in this file need to be replaced with `Regular10`s. If 10 isn't in this branch yet, please leave a prototype.